docs/how_to_guides/how_to_create_surrogate_models.rst

```diff
@@ -0,0 +1,168 @@
+  from idaes.core.surrogate.pysmo_surrogate import PysmoRBFTrainer, PysmoSurrogate
+  from idaes.core.surrogate.surrogate_block import SurrogateBlock
+
+Define the surrogate model class and required parameters. The model class should contain the following:
+
+* build method that defines the surrogate inputs, outputs, and their respective bounds
+* get_training_validation method that loads the training and validation data
+* create_surrogate method that creates the surrogate model, provides the training options, and saves the resulting model
+* load_surrogate method that loads the surrogate model
+
+
+.. testcode::
+
+  @declare_process_block_class("PVSurrogate")
+  class PVSurrogateData(SolarEnergyBaseData):
+    """
+    Surrogate model for PV.
+    """
+
+    CONFIG = SolarEnergyBaseData.CONFIG()
+
+    def build(self):
+        super().build()
+
+        self._tech_type = "PV"
+
+        self.design_size = Var(
+            initialize=1000,
+            bounds=[1, 200000],
+            units=pyunits.kW,
+            doc="PV design size in kW",
+        )
+
+        self.annual_energy = Var(
+            initialize=1,
+            units=pyunits.kWh,
+            doc="annual energy produced by the plant in kWh",
+        )
+        
+        self.land_req = Var(
+            initialize=7e7,
+            units=pyunits.acre,
+            doc="annual energy produced by the plant in kWh",
+        )
+
+        self.surrogate_inputs = [self.design_size]
+        self.surrogate_outputs = [self.annual_energy, self.land_req]
+
+        self.input_labels = ["design_size"]
+        self.output_labels = ["annual_energy", "land_req"]
+
+        self.electricity_constraint = Constraint(
+            expr=self.annual_energy
+            == -1 * self.electricity
+            * pyunits.convert(1 * pyunits.year, to_units=pyunits.hour)
+        )
+
+Create Training Data
+
+.. testcode::
+
+  def get_training_validation(self):
+    self.dataset_filename = os.path.join(
+        os.path.dirname(__file__), "data/dataset.pkl"
+    )
+    print('Loading Training Data...\n')
+    time_start = time.process_time()
+    pkl_data = pd.read_pickle(self.dataset_filename)
+    data = pkl_data.sample(n=int(len(pkl_data))) #FIX default this to 100% of data
+    self.data_training, self.data_validation = split_training_validation(
+        data, self.training_fraction, seed=len(data)
+    )
+    time_stop = time.process_time()
+    print("Data Loading Time:", time_stop - time_start, "\n")
+
+Create Surrogate
+
+.. testcode::
+
+  def create_surrogate(self):
+    self.training_fraction = 0.8 # Fraction of the sampled data to split for training and validation
+
+    self.get_training_validation()
+    time_start = time.process_time()
+
+    # Create PySMO trainer object
+    trainer = PysmoRBFTrainer(
+        input_labels=self.input_labels,
+        output_labels=self.output_labels,
+        training_dataframe=self.data_training,
+    )
+
+  # Set PySMO options
```

Comment:
starting here it looks like you are outdented one level; i.e. the rest of this code block should be indented one more to fit under ``create_surrogate``

Reply:
Good catch